pkg/server/registry/images/strategy.go

```diff
@@ -3,177 +3,156 @@ package images
-	if err != nil {
-		return apiv1.ImageList{}, err
 	}
+	imageList := &apiv1.ImageList{}
 
-	repo, err := getRepo(ns)
-	if err != nil {
-		return apiv1.ImageList{}, err
-	}
-
-	names, err := remote.List(repo, opts...)
-	if tErr, ok := err.(*transport.Error); ok && tErr.StatusCode == http.StatusNotFound {
-		return apiv1.ImageList{}, nil
-	}
-	if err != nil {
-		return apiv1.ImageList{}, err
-	}
-
-	tags, err := tags2.Get(ctx, s.client, ns)
+	err := s.client.List(ctx, imageList, &kclient.ListOptions{
+		Namespace: image.Namespace,
+	})
 	if err != nil {
-		return apiv1.ImageList{}, err
+		result = append(result, field.InternalError(field.NewPath("namespace"), err))
 	}
 
-	result := apiv1.ImageList{}
-	for _, imageName := range names {
-		if !tags2.SHAPattern.MatchString(imageName) {
+	for _, imageItem := range imageList.Items {
+		if imageItem.Digest == image.Digest {
 			continue
 		}
-		tags := tags[imageName]
-		if len(tags) == 0 {
-			tags = append(tags, "")
-		}
-		for _, tag := range tags {
-			image := apiv1.Image{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      imageName,
-					Namespace: ns,
-				},
-				Digest:    "sha256:" + imageName,
-				Reference: tag,
+		for i, tag := range imageItem.Tags {
+			currentTag := imageItem.Tags[i]
+			if duplicateTag[imageItem.Tags[i]] {
+				result = append(result, field.Duplicate(field.NewPath("tag name"), fmt.Errorf("unable to tag image %s with tag %s as it is already in use by %s", image.Name[:12], tag, imageItem.Name[:12])))
 			}
-			if tag != "" {
-				parsedTag, err := name.NewTag(tag)
-				if err == nil {
-					image.Repository = strings.TrimSuffix(tag, ":"+parsedTag.TagStr())
-					image.Tag = parsedTag.TagStr()
-				}
-			}
-			result.Items = append(result.Items, image)
+			duplicateTag[currentTag] = true
 		}
 	}
+	return result
+}
 
-	return result, nil
+func (s *Strategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) (result field.ErrorList) {
+	return s.Validate(ctx, obj)
 }
 
-func (s *Strategy) Delete(ctx context.Context, obj types.Object) (types.Object, error) {
-	image, matchedReference, err := s.imageGet(ctx, obj.GetNamespace(), obj.GetName())
-	if err != nil {
-		return nil, err
-	}
+func (s *Strategy) ConvertToTable(ctx context.Context, object runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
+	return tables.ImageConverter.ConvertToTable(ctx, object, tableOptions)
+}
+func getRepo(namespace string) (name.Repository, error) {
+	return name.NewRepository("127.0.0.1:5000/acorn/" + namespace)
+}
+func (s *Strategy) Get(ctx context.Context, namespace, name string) (types.Object, error) {
+	return s.ImageGet(ctx, namespace, name)
+}
 
-	if matchedReference != "" {
-		tagCount, err := tags2.Remove(ctx, s.client, image.Namespace, image.Digest, matchedReference)
-		if tagCount > 0 || err != nil {
-			return image, nil
+func (s *Strategy) Create(ctx context.Context, obj types.Object) (types.Object, error) {
```

Comment:
After darren's refactor, this create method should completely go away, since users wont be allowed to Create images, just ImageInstances should be created by the backend

Reply:
added a TODO